nntrainer/tensor/cl_operations/testing_rotarty_emb.cpp

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Yash Singh <yash.singh@samsung.com>
+ *
+ * @file	testing_rotary_emb.cpp
+ * @date	28 August 2024
+ * @brief	Rotary Embedding CPU code
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Yash Singh <yash.singh@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#include "tensor.h"
+#include <string>
+
+/**
+ * @brief     compute frequency for rotary embedding
+ * @param[in] dim hidden dim size
+ * @param[in] seq_len sequency length
+ * @param[out] freqs_cos cosine of the frequencies
+ * @param[out] freqs_sin sine of the frequencies
+ * @param[out] freqs base frequencies array to be used in computation of cos and
+ * sin values for each position in sequence
+ * @param[in] theta rotary angle
+ */
+void precompute_freqs(int dim, unsigned int seq_len,
+                      std::vector<std::vector<float>> &freqs_cos,
+                      std::vector<std::vector<float>> &freqs_sin,
+                      std::vector<float> &freqs, float theta = 10000.0) {
+  if (freqs_cos.empty()) {
+    unsigned int half_ = dim / 2;
+    for (unsigned int i = 0; i < half_; ++i) {
+      freqs.push_back(1.0 /
+                      (std::pow(theta, (2 * i) / static_cast<float>(dim))));
+    }
+
+    auto cos = std::vector<std::vector<float>>();
+    cos.assign(seq_len, std::vector<float>(dim, 0));
+
+    auto sin = std::vector<std::vector<float>>();
+    sin.assign(seq_len, std::vector<float>(dim, 0));
+
+    for (unsigned int i = 0; i < seq_len; ++i) {
+#ifdef USE_NEON
+      nntrainer::calc_trigonometric_vals_dup(half_, freqs.data(), cos[i].data(),
+                                             sin[i].data(), i);
+#else
```

Comment:
Is using NEON during cl operations possible?

Reply:
No, actually I took this code for rotary emb from the felice repo. This file was just to test rotary emb on the cpu and to run the unittests. Regardless of it, I've updated and removed this code.